### PR TITLE
Faraz/lp 2136

### PR DESCRIPTION
**Pull request by m-ali-0001**
### Description

[LP-](https://philanthropyu.atlassian.net/browse/LP-)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/###)

Add a description of your changes here.

### Notes

skip ci

### CODE REVIEW

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: requirements/philu/testing.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# TODO: Find replace of add_or_replace_params method with urllib or some some django builtin package and remove this dependency
```
Comment (harris-irfan) [inline requirements/philu/testing.txt:1]: Find replacement...

File: scripts/Jenkinsfiles/python-unittest [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    environment {
+        XDIST_GIT_BRANCH = xdist_git_branch()
+        XDIST_GIT_REFSPEC = xdist_git_refspec()
+        XDIST_INSTANCE_TYPE = "m5.large"
+        XDIST_WORKER_AMI = credentials('XDIST_WORKER_AMI')
+        XDIST_WORKER_IAM_PROFILE_ARN = credentials('XDIST_WORKER_IAM_PROFILE_ARN')
+        XDIST_WORKER_KEY_NAME = "philu-jenkins-worker"
+        XDIST_WORKER_SUBNET = credentials('XDIST_WORKER_SUBNET')
+        XDIST_WORKER_SECURITY_GROUP = credentials('XDIST_WORKER_SECURITY_GROUP')
+        WTW_CONTEXT = "python"
```
Comment (harris-irfan) [inline scripts/Jenkinsfiles/python-unittest:55]: can we avoid double qoutes, here and elsewhere in this file?

File: scripts/Jenkinsfiles/python-unittest [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+                            string(name: 'DESCRIPTION', value: 'Pending'),
+                            string(name: 'CONTEXT', value: "${GITHUB_CONTEXT}"),
+                            string(name: 'CREATE_DEPLOYMENT', value: 'false'),
+                            string(name: 'BUILD_STATUS', value: 'pending')
+                        ],
+                        propagate: false, wait: false
+                }
+            }
+        }
+        stage("lms-unit") {
```
Comment (harris-irfan) [inline scripts/Jenkinsfiles/python-unittest:82]: double qoutes to be replaced with single qoutes

File: scripts/Jenkinsfiles/python-unittest [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                    // For master jobs run github-build-status and report to slack when necessary
+                    if (currentBuild.currentResult == "SUCCESS") {
+                        create_deployment = "true"
+                        build_status = "success"
+                        build_description = "Build Passed"
+                    }
+                    else {
+                        create_deployment = "false"
+                        build_status = "failure"
+                        build_description = "Build Failed"
```
Comment (harris-irfan) [inline scripts/Jenkinsfiles/python-unittest:170]: single qoutes?

File: scripts/xdist/pytest_worker_manager.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+                break
+
+        if not all_running:
+            raise Exception(
+                "Timed out waiting to spin up all workers."
+            )
+        logger.info("Successfully booted up {} workers.".format(number_of_workers))
+
+        not_ready_ip_addresses = ip_addresses[:]
+        logger.info("Checking ssh connection to workers.")
```
Comment (harris-irfan) [inline scripts/xdist/pytest_worker_manager.py:124]: can we shift to single qoutes here, line 119, line 121 and elsewhere in this file?

File: pavelib/prereqs.py [hunk 6]
```diff
@@ -31,7 +31,11 @@
 
 # Developers can have private requirements, for local copies of github repos,
 # or favorite debugging tools, etc.
-PRIVATE_REQS = 'requirements/philu/base.txt'
+if 'TOXENV' in os.environ:
+    PRIVATE_REQS = 'requirements/philu/testing.txt'
+else:
+    PRIVATE_REQS = 'requirements/philu/base.txt'
+
```
Comment (harris-irfan) [inline pavelib/prereqs.py:38]: Can we use ternary operator here?

`PRIVATE_REQS = 'requirements/philu/testing.txt' if 'TOXENV' in os.environ else 'requirements/philu/base.txt'`

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkin run all

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: scripts/unit-tests.sh [hunk 7]
```diff
@@ -126,4 +137,19 @@ case "${TEST_SUITE}" in
...
+        
+        philu_apps_list=(
+            "openedx/features/marketplace/"
+        )
+        paver test_system -s lms -t openedx/features/marketplace/ -v --processes=1 2> philu-tests.log
+        # philu_apps_str="${philu_apps_list[@]}"
+        # paver test_system -s lms -t "${philu_apps_str}" ${PAVER_ARGS} ${PARALLEL} 2> philu-tests.log
+        mv reports/.coverage reports/.coverage.philu
 esac
+-v --processes=1
```
Comment (zahid-arbisoft) [inline scripts/unit-tests.sh:155]: EOF here

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by m-ali-0001** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

File: cms/envs/test.py [hunk 8]
```diff
@@ -332,20 +332,16 @@
...
-CONFIG_ROOT = path(os.environ.get('CONFIG_ROOT', ENV_ROOT))
-
-CONFIG_PREFIX = SERVICE_VARIANT + "." if SERVICE_VARIANT else ""
-
-with open(CONFIG_ROOT / CONFIG_PREFIX + "auth.json") as auth_file:
-    AUTH_TOKENS = json.load(auth_file)
-
+##################### PhilU Settings #######################
 NODEBB_RETRY_DELAY = 60
 NODEBB_ENDPOINT = "http://local.philanthropyu.org:4567"
```
[the remark this answers is not in the document]
Comment (ahmed-faraz46) [reply cms/envs/test.py:337]: Done!

File: requirements/philu/testing.txt (hunk 1 above)
Comment (ahmed-faraz46) [reply requirements/philu/testing.txt:1]: Done!

File: scripts/Jenkinsfiles/python-unittest (hunk 3 above)
Comment (ahmed-faraz46) [reply scripts/Jenkinsfiles/python-unittest:88]: Done!

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
jenkins run all

File: requirements/philu/testing.txt [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# TODO: Find replacement of add_or_replace_params method with urllib or some some django builtin package and remove this dependency
+w3lib==1.21.0
+-e git+https://github.com/philanthropy-u/pynodebb.git#egg=pynodebb
+SurveyGizmo==1.2.3
+mandrill==1.0.57
+pyminizip==0.2.1
+channels==1.1.8
+asgi-redis==1.4.3
+imgkit==1.0.1
+pdfkit==0.6.1
```
Comment (zahid-arbisoft) [inline requirements/philu/testing.txt:10]: Ideally, we should mock third-party apps like imgkit, pdfkit, and mandrill, etc. We can remove it in the future, once all test are mocked properly

File: scripts/jenkins-common.sh [hunk 10]
```diff
@@ -24,20 +24,20 @@ fi
...
 # Reset the jenkins worker's virtualenv back to the
 # state it was in when the instance was spun up.
-if [ -e $HOME/edx-venv_clean.tar.gz ]; then
-    rm -rf $HOME/edx-venv
-    tar -C $HOME -xf $HOME/edx-venv_clean.tar.gz
-fi
+# if [ -e $HOME/edx-venv_clean.tar.gz ]; then
+#     rm -rf $HOME/edx-venv
+#     tar -C $HOME -xf $HOME/edx-venv_clean.tar.gz
+# fi
```
Comment (zahid-arbisoft) [inline scripts/jenkins-common.sh:30]: Should we keep commented code

File: scripts/jenkins-common.sh [hunk 11]
```diff
@@ -24,20 +24,20 @@ fi
...
-if [ -e $HOME/edx-npm-cache_clean.tar.gz ]; then
-    echo "Loading archived npm packages into the local npm cache"
-    rm -rf $HOME/.npm
-    tar -C $HOME -xf $HOME/edx-npm-cache_clean.tar.gz
-fi
+# if [ -e $HOME/edx-npm-cache_clean.tar.gz ]; then
+#     echo "Loading archived npm packages into the local npm cache"
+#     rm -rf $HOME/.npm
+#     tar -C $HOME -xf $HOME/edx-npm-cache_clean.tar.gz
+# fi
```
Comment (zahid-arbisoft) [inline scripts/jenkins-common.sh:40]: same

File: scripts/unit-tests.sh [hunk 12]
```diff
@@ -50,9 +50,20 @@ else
...
 
+# declare -a test_suites=(
+# "openedx/features/marketplace/tests.py"
+# "openedx/features/idea/tests.py"
+# )
+
+# for val in ${test_suites[@]}; do
+#     paver test_system -s lms -t $val ${PAVER_ARGS} ${PARALLEL} 2> lms-tests.log
+# done
+# mv reports/.coverage reports/.coverage.lms
```
Comment (zahid-arbisoft) [inline scripts/unit-tests.sh:61]: here as well

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by philanthropyu** [comment]
Your PR has finished running tests. The following contexts failed:
* jenkins/python

**Comment by ahmed-faraz46** [comment]
Code is merged under [this PR](https://github.com/philanthropy-u/edx-platform/pull/1724)

**Pull request closed without merging**